flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/KinesisConfigUtil.java

```diff
@@ -171,7 +172,14 @@ public static void validateAwsConfiguration(Properties config) {
 		}
 
 		if (!config.containsKey(AWSConfigConstants.AWS_REGION)) {
```

Comment:
I wonder if its actually sufficient to rely on this check, without having the need to have a bunch of the new constructors that you added.

I.e., it is sufficient that the user doesn't supply a `AWSConfigConstants.AWS_REGION` value in the props, which we would then default to using `Regions.getCurrentRegion()`.